### ISSUE TITLE
Deleting an object that was detected as leaked

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1333]
The code below eventually results in the error mentioned in the issue title.

```
			int zero = 0;
			for (int i = 0; i < 1000000; i++)
				System.Threading.ThreadPool.QueueUserWorkItem(new () => {
					Console.WriteLine(zero);
				});
```
Tested with https://github.com/beefytech/Beef/commit/f35a3de175065572bdfc1f9f819e1549dcbc4977

**Comment by bfiete** [comment]
Fixed at 14e1ffa5ac62fab24622af245bfad8e471de5175

The core issue was that the work queue is implemented using a List with work items popped off the front, which requires a lot of memmove activity (it should use a deque). There are times during memmove where some list data is only available in the 256-bit ymm registers, and we weren't previously scanning those for object references.

**Comment by bfiete** [comment]
So this would trigger even if you just had a huge list of strings that you continuously popped and deleted items off the front of.
